qemu/tests/balloon_check.py

```diff
@@ -107,6 +107,10 @@ def balloon_memory(self, new_mem):
             if self.params.get('illegal_value_check', 'no') == 'no' and new_mem != self.get_ballooned_memory():
                 raise exceptions.TestFail("Balloon memory fail with error"
                                           " message: %s" % e)
+
+        balloon_timeout = float(self.params.get("balloon_timeout", 240))
+        self.wait_for_balloon_complete(balloon_timeout)
```

Comment:
Could you please explain why it need to be called at this?

Reply:
After balloon operation, we should wait until guest memory do not change, then do memory compare.
Else it may cause memory compare fail as guest memory still changes.

Reply:
I saw there is another check at line 125, are they doing the same thing? Looks like you just need another `240s` (`balloon_timeout`).

Reply:
In my understanding, line 125 is checking mem only from monitor, not mem in guest.
Correct me if I'm wrong.